src/morphosamplers/spline.py

```diff
@@ -53,140 +49,119 @@ def validate_number_of_points(
+        u = splev([np.asarray(u)], self._equidistant_spline_tck)
         samples = splev(u, self._raw_spline_tck, der=derivative_order)
-        return einops.rearrange(samples, "c 1 1 b -> b c")
+        return np.atleast_2d(np.squeeze(samples).T)
 
-    def _get_equidistance_u(self, separation: float) -> np.ndarray:
-        """Get equally spaced values of u.
-
-        Parameters
-        ----------
-        separation : float
-            The distance between the u values in euclidian distance.
-
-        Returns
-        -------
-        u : np.ndarray
-            The array of equally-spaced spline coordinates..
-        """
+    def _get_equidistant_u(self, separation: float) -> np.ndarray:
+        """Get values for u which yield spline samples with a defined Euclidean separation."""
         n_points = int(self._length // separation)
         remainder = (self._length % separation) / self._length
         return np.linspace(0, 1 - remainder, n_points)
 
-    def _get_equidistance_spline_samples(
-        self, separation: float, derivative_order: bool = 0
+    def _get_equidistant_spline_samples(
+            self, separation: float, derivative_order: int = 0
     ) -> np.ndarray:
-        """Calculate equidistant spline samples with a defined separation.
-
-        Parameters
-        ----------
-        sepration : float
-            The distance between points in euclidian space.
-        derivative_order : int
-            Order of the derivative to evaluate at each spline position.
-            If 0, the position on the spline is returned.
-            If >0, the derivative of position is returned (e.g., 1 for tangent vector).
-            derivative_order must be <= the spline order.
-            Default value is 0.
-
-        Returns
-        -------
-        values : np.ndarray
-            The values along the spline.
-            If derivative_order is 0, returns positions.
-            If calculate_derivative >0, returns derivative vectors.
-        """
-        if (derivative_order < 0) or (derivative_order > self.order):
-            # derivative order must be 0 < derivative_order < spline_order
-            raise ValueError("derivative order must be [0, spline_order]")
-        u = self._get_equidistance_u(separation)
+        """Calculate equidistant spline samples with a defined separation in Euclidean space."""
+        u = self._get_equidistant_u(separation)
         return self.sample_spline(u, derivative_order=derivative_order)
+
+
+class Spline3D(NDimensionalSpline):
+    """3D spline model with interpolation of orientations along the spline.
+
+    Basis vectors for a local coordinate system along the spline can be calculated.
+    In this coordinate system, z vectors are tangent to the spline and the xy-plane
+    changes minimally and smoothly.
+    """
+    _rotation_sampler = PrivateAttr(Slerp)
+
+    @validator('points')
+    def _is_n_plus_3d_coordinate_array(cls, v):
```

Comment:
```suggestion
    def _is_n_by_3_coordinate_array(cls, v):
```

how about the above if you didn't like `is_3d_coordinate_array`? n plus three doesn't make much sense to me

Reply:
Oops, this was aleftover from trying to do `n+3d` handling.